legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -47,9 +51,15 @@ def validate(incorporation_json: Dict):
     if err:
         msg.extend(err)
 
-    err = validate_share_structure(incorporation_json, coreFiling.FilingTypes.INCORPORATIONAPPLICATION.value)
-    if err:
-        msg.extend(err)
+    if legal_type == Business.LegalTypes.BCOMP.value:
+        err = validate_share_structure(incorporation_json, coreFiling.FilingTypes.INCORPORATIONAPPLICATION.value)
+        if err:
+            msg.extend(err)
+
+    if legal_type == Business.LegalTypes.COOP.value:
+        err = validate_cooperative_documents(incorporation_json)
+        if err:
+            msg.extend(err)
 
```

Comment:
maybe we should `elsif` here in preparation for other legal types.

Reply:
For sure, will do!